public/app/features/dashboard/components/ShareModal/SharePublicDashboard/Configuration.tsx

```diff
@@ -36,6 +40,19 @@ export const Configuration = ({
             <Label description="The public dashboard uses the default time settings of the dashboard">Time range</Label>
             <TimeRangeInput value={timeRange} disabled onChange={() => {}} />
           </Layout>
+          <Layout orientation={isDesktop ? 0 : 1} spacing="xs" justify="space-between">
+            <Label description="Show annotations on public dashboard">Show Annotations</Label>
+            <Switch
+              data-testid={selectors.EnableAnnotationsSwitch}
+              value={isAnnotationsEnabled}
+              onChange={() => {
+                reportInteraction('grafana_dashboards_annotations_toggled', {
```

Comment:
Nice! The name of this action event should be `clicked` instead of `toggled`

```suggestion
                reportInteraction('grafana_dashboards_annotations_clicked', {
```